### ISSUE TITLE
ApplySpec not properly handling properties with arrays of anonymous objects

### ISSUE THREAD

**Issue by ryanbas21** [ramda/ramda#2784]
```
applySpec({
  test: [{ a : inc }]
})(2);

// {"test": {"0": {"a": 3}}} test should have a value of an array, not of an object.
```
https://ramdajs.com/repl/#?applySpec%28%7B%0A%20%20test%3A%20%5B%7B%20a%20%3A%20inc%20%7D%5D%0A%7D%29%282%29%3B%0A

What is happening is that when applySpec recursively goes into the array, it changes the array to an object with a key "0" and then applies the function, where previously it handled the array appropriately. 

I am seeing this issue in the new `0.26.1`/ `0.26` and not 0.25.

**Comment by CrossEye** [comment]
Hmm, I wonder what caused this regression.  I don't recall any recent work on `applySpec`.  I'll have to check this weekend.

**Comment by ryanbas21** [comment]
@CrossEye The only thing we noticed was a PR merged about the `map` key with applySpec, although I could not tell how that would affect this regression

**Comment by ryanbas21** [comment]
Actually, that function that was added is the issue. ITs taking the keys and not retransforming it back to an array

**Comment by CrossEye** [comment]
#2681 #2682

**Comment by bensbigolbeard** [comment]
I've submitted a potential fix for this issue (☝️)

@CrossEye Let me know if there are any issues with this approach.

**Comment by jordie23** [comment]
I posted this in another issue before I noticed this, so reposting here because I think it applies, however my issue was the root object being an array not necessarily a nested property.

Original comment:

------------------------

Hello,

So turns out, previous to this, `applySpec` worked seemlessly with arrays/lists, not just objects. Since this change, it has broken that. Just wondering whether `applySpec` should be supporting arrays/lists or whether that was accidental previously? This has broken some of our code - not a problem, we can update it - but just wondering whether we were relying on a quirk in the code or something that was actually supposed to be supported?

e.g.

```const kvs = [
  {
    Key: 'Tag1',
    Value: 'TestVal1'
  },
  {
    Key: 'Tag2',
    Value: 'TestVal2'
  },
  {
    Key: 'Tag3',
    Value: 'TestVal3'
  }
]

applySpec([map(prop('Key')), map(prop('Value'))])(kvs)

// result < 0.26

[ 
  [ 'Tag1', 'Tag2', 'Tag3', 'Tag4' ],
  [ 'TestVal1', 'TestVal2', 'TestVal3', 'TestVal4' ]
]
  
// result >= 0.26

{
  "0": [ 'Tag1', 'Tag2', 'Tag3', 'Tag4' ],
  "1": [ 'TestVal1', 'TestVal2', 'TestVal3', 'TestVal4' ]
}
```

**Comment by bensbigolbeard** [comment]
@jordie23 you are correct, this issue does also include using an array at the base level. I added a test case to the currently open PR and the change I made also addresses your scenario

**Comment by bensbigolbeard** [comment]
Ok, i added a test case to validate this use-case as well @jordie23. you can check out the PR ☝️ referenced above.
